src/ducks/sharing/index.js

```diff
@@ -0,0 +1,181 @@
+import styles from './share.styl'
+
+import React from 'react'
+import CopyToClipboard from 'react-copy-to-clipboard'
+import Modal, { ModalContent } from 'cozy-ui/react/Modal'
+import Toggle from 'cozy-ui/react/Toggle'
+import classnames from 'classnames'
+import { Tab, Tabs, TabList, TabPanels, TabPanel } from 'cozy-ui/react/Tabs'
+// import Alerter from 'cozy-ui/react/Alerter'
+
+/* global cozy */
+export const filterSharedDocuments = (ids, doctype) =>
+  findPermSets(ids, doctype).then(sets => sets.map(set => set.attributes.permissions.collection.values[0]))
+
+export const findPermSet = (id, doctype) =>
+  findPermSets([id], doctype).then(sets => sets.length === 0 ? undefined : sets[0])
+
+// TODO: move this to cozy-client-js
+// there is cozy.client.files.getCollectionShareLink already, but I think that
+// there is a need of a bit of exploratory work and design for that API...
+
+export const findPermSets = (ids, doctype) =>
+  cozy.client.fetchJSON('GET', `/permissions/doctype/${doctype}/sharedByLink`)
+    .then(sets => sets.filter(set => {
+      const perm = set.attributes.permissions.collection
+      return perm.type === doctype && ids.find(id => perm.values.indexOf(id) !== -1) !== undefined
+    }))
+
+export const createPermSet = (id, doctype) =>
+  cozy.client.fetchJSON('POST', `/permissions?codes=email`, {
+    data: {
+      type: 'io.cozy.permissions',
+      attributes: {
+        permissions: {
+          files: {
+            type: 'io.cozy.files',
+            verbs: ['GET'],
+            values: [id],
+            selector: 'referenced_by'
+          },
+          collection: {
+            type: doctype,
+            verbs: ['GET'],
+            values: [id]
+          }
+        }
+      }
+    }
+  })
```

Comment:
I faced a `401` when using the generated link, this code will be updated as soon as I get answers (see IRC#back)